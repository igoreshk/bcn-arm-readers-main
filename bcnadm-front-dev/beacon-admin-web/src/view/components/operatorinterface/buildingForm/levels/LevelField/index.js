import React, { Component } from 'react';
import { TextField, Button } from '@material-ui/core';
import PropTypes from 'prop-types';
import { maxFileSize } from 'src/view/FileConsts';
import LevelImageDropZone from '../LevelImageDropZone';
import RemoveLevelDialog from '../RemoveLevelDialog';
const multiplier = 5;
const maxLevelImageMapSize = multiplier * maxFileSize; // max level image size

export default class LevelField extends Component {
  changeCurrentLevelNumber = (event) => {
    const {
      changeLevelNumber,
      params: { levelIndex },
      input: { onChange, value }
    } = this.props;
    const newNumber = Number(event.target.value);
    onChange({
      ...value,
      entityId: value.entityId,
      number: newNumber
    });
    changeLevelNumber(levelIndex, newNumber);
  };

  onRemoveFieldClick() {
    const {
      onRemoveLevelClick,
      params: { levelIndex }
    } = this.props;
    onRemoveLevelClick(levelIndex);
  }

  changeCurrentLevelImage = (newImage) => {
    const {
      params: { levelIndex },
      changeLevelImage
    } = this.props;
    changeLevelImage(levelIndex, newImage);
  };

  renderButton = () => {
    const {
      params: { number, entityId },
      translate
    } = this.props;
    return entityId ? (
      <RemoveLevelDialog
        levelId={entityId}
        levelName={number}
        onRemoveLevelClick={() => this.onRemoveFieldClick()}
        translate={translate}
      />
    ) : (
      <Button onClick={() => this.onRemoveFieldClick()} classes={{ root: 'removeButton', label: 'removeButtonText' }}>
        {translate('removeLevel.removeButton')}
      </Button>
    );
  };

  render() {
    const {
      params: { entityId, number, imageLink, image, unique },
      input,
      translate
    } = this.props;

    return (
      <div className="createLevelContainer">
        <TextField
          type="number"
          name="level"
          value={number}
          onChange={this.changeCurrentLevelNumber}
          className="levelNumber"
          error={!unique}
          helperText={unique ? null : translate('validation.uniqueLevel')}
        />
        <LevelImageDropZone
          imageLink={imageLink}
          maxSize={maxLevelImageMapSize}
          input={input}
          entityId={entityId}
          number={number}
          image={image}
          changeLevelImage={this.changeCurrentLevelImage}
          translate={translate}
        />
        {this.renderButton()}
      </div>
    );
  }
}

LevelField.propTypes = {
  input: PropTypes.shape({
    image: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
    mimeType: PropTypes.string,
    entityId: PropTypes.string,
    onChange: PropTypes.func
  }).isRequired,
  params: PropTypes.shape({
    entityId: PropTypes.string,
    image: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
    number: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
    imageLink: PropTypes.string,
    unique: PropTypes.bool
  }).isRequired,
  translate: PropTypes.func.isRequired,
  changeLevelNumber: PropTypes.func,
  onRemoveLevelClick: PropTypes.func,
  changeLevelImage: PropTypes.func
};
