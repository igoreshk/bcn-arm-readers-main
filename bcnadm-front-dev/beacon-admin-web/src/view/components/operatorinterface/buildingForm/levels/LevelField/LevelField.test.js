import React from 'react';
import { shallow } from 'enzyme';

import LevelField from './index';

const props = {
  input: {
    image: 'image',
    mimeType: 'mimeType',
    entityId: 'entityId',
    value: {
      image: 'image',
      number: 1,
      imageLink: 'imageLink'
    },
    onChange: jest.fn()
  },
  params: {
    imageLink: 'imageLink',
    number: 0,
    image: undefined
  },
  translate: jest.fn(),
  onRemoveLevelClick: jest.fn(),
  changeLevelNumber: jest.fn(),
  changeLevelImage: jest.fn()
};

const wrapper = shallow(<LevelField {...props} />);

describe('LevelField test', () => {
  it('check props matches', () => expect(wrapper.length).toEqual(1));
  it('snapshot for LevelField', () => expect(wrapper).toMatchSnapshot());
});
