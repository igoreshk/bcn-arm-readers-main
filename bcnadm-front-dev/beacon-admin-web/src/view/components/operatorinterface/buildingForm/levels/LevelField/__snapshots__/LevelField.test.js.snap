// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LevelField test snapshot for LevelField 1`] = `
<div
  className="createLevelContainer"
>
  <WithStyles(ForwardRef(TextField))
    className="levelNumber"
    error={true}
    name="level"
    onChange={[Function]}
    type="number"
    value={0}
  />
  <LevelImageDropZone
    changeLevelImage={[Function]}
    imageLink="imageLink"
    input={
      Object {
        "entityId": "entityId",
        "image": "image",
        "mimeType": "mimeType",
        "onChange": [MockFunction],
        "value": Object {
          "image": "image",
          "imageLink": "imageLink",
          "number": 1,
        },
      }
    }
    maxSize={5242880}
    number={0}
    translate={
      [MockFunction] {
        "calls": Array [
          Array [
            "validation.uniqueLevel",
          ],
          Array [
            "removeLevel.removeButton",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
  />
  <WithStyles(ForwardRef(Button))
    classes={
      Object {
        "label": "removeButtonText",
        "root": "removeButton",
      }
    }
    onClick={[Function]}
  />
</div>
`;
